kate/grid/src/dims.rs

```diff
@@ -0,0 +1,104 @@
+use core::num::NonZeroUsize;
+
+/// The dimensions of a grid
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Dimensions {
+	width: NonZeroUsize,
```

Comment:
Could we use `NonZeroU32` instead? That will produce the same underlying types for `Native` & `Wasm`.